src/shogun/base/SGObject.h

```diff
@@ -1011,5 +1021,64 @@ class CSGObject
 	/** Subscriber used to call onNext, onComplete etc.*/
 	SGSubscriber* m_subscriber_params;
 };
+
+#ifndef SWIG
```

Comment:
could we move this whole thing into a separate header... as sgboject.h is already 1k+ LoC

Reply:
Yes good point